web/src/search/input/QueryInput.tsx

```diff
@@ -244,93 +203,151 @@ export class QueryInput extends React.Component<Props, State> {
         this.subscriptions.unsubscribe()
     }
 
-    public componentDidUpdate(prevProps: Props, prevState: State): void {
+    public componentDidUpdate(_: Props, prevState: State): void {
+        if (prevState.cursorPosition !== this.state.cursorPosition) {
+            this.focusInputAndPositionCursor(this.state.cursorPosition)
+        }
         this.componentUpdates.next(this.props)
-        // Check if selected suggestion is out of view
-        scrollIntoView(this.suggestionListElement, this.selectedSuggestionElement)
     }
 
     public render(): JSX.Element | null {
-        const showSuggestions =
-            this.props.value.length >= QueryInput.SUGGESTIONS_QUERY_MIN_LENGTH &&
-            this.state.inputFocused &&
-            !this.state.hideSuggestions &&
-            this.state.suggestions.length !== 0
+        const showSuggestions = !!this.state.suggestions.values.length
```

Comment:
Nit: this reads clearer, because `!!`/`!` is generally used for testing for the _presence_ of a property
```suggestion
        const showSuggestions = this.state.suggestions.values.length > 0
```